addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -56,7 +56,10 @@ options.registry.TableOfContent = options.Class.extend({
             $el.attr('id', id);
             $el[0].dataset.anchor = 'true';
         });
-        $nav.find('a:first').addClass('active');
+        const targetNode = this.$target.find('.s_table_of_content_main')[0]
+        targetNode.children.length === 0
+            ? (this.trigger_up('go_to_parent', {$snippet: this.$target}), this.$target.remove())
+            : $nav.find('a:first').addClass('active');
```

Comment:
When it's not an assignation, we encourage if else instead of ternary operator :+1: